ui/DesignSystem/Component/SourceBrowser/RevisionSelector.svelte

```diff
@@ -1,19 +1,142 @@
 <script>
   import { createEventDispatcher } from "svelte";
 
-  import { Input } from "../../Primitive";
+  import { Avatar, Icon } from "../../Primitive";
 
+  export let currentRevision = null;
+  export let expanded = false;
   export let revisions = null;
+  export let style = "";
 
-  let revision;
+  // Dropdown element. Set by the view.
+  let dropdown = null;
+
+  const showDropdown = () => {
+    expanded = true;
+  };
+
+  const hideDropdown = () => {
+    expanded = false;
+  };
+
+  const handleClick = ev => {
+    // Any click *outside* the dropdown should hide hte dropdown.
```

Comment:
I feel like my review starts hitting diminishing returns ; P

```suggestion
    // Any click *outside* the dropdown should hide the dropdown.
```